server/datastore/mysql/apple_mdm.go

```diff
@@ -1674,6 +1674,71 @@ func (ds *Datastore) GetMDMAppleBootstrapPackageBytes(ctx context.Context, token
 	return &bp, nil
 }
 
+// TODO(Sarah): Use constants to map ncr.status to bootstrap package status according to Apple
+// specs. https://developer.apple.com/documentation/devicemanagement/installenterpriseapplicationresponse
+func (ds *Datastore) GetMDMAppleBootstrapPackageSummary(ctx context.Context, teamID uint) (*fleet.MDMAppleBootstrapPackageSummary, error) {
+	stmt := `
+          SELECT
+              COUNT(IF(ncr.status = 'Acknowledged', 1, NULL)) AS installed,
+              COUNT(IF(ncr.status = 'Error', 1, NULL)) AS failed,
+              COUNT(IF(ncr.status IS NULL OR (ncr.status != 'Acknowledged' AND ncr.status != 'Error'), 1, NULL)) AS pending
+          FROM
+              hosts h
+          LEFT JOIN host_mdm_apple_bootstrap_packages hmabp ON
+              hmabp.host_uuid = h.uuid
+          LEFT JOIN nano_command_results ncr ON
+              ncr.command_uuid  = hmabp.command_uuid
+          JOIN host_mdm hm ON
+              hm.host_id = h.id
+          WHERE
+              hm.installed_from_dep = 1 AND COALESCE(h.team_id, 0) = ?`
+
+	var bp fleet.MDMAppleBootstrapPackageSummary
+	if err := sqlx.GetContext(ctx, ds.reader, &bp, stmt, teamID); err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "get bootstrap package summary")
+	}
+	return &bp, nil
+}
+
+func (ds *Datastore) RecordHostBootstrapPackage(ctx context.Context, commandUUID string, hostUUID string) error {
+	stmt := "INSERT INTO host_mdm_apple_bootstrap_packages (command_uuid, host_uuid) VALUES (?, ?)"
+	_, err := ds.writer.ExecContext(ctx, stmt, commandUUID, hostUUID)
+	return ctxerr.Wrap(ctx, err, "record bootstrap package command")
+}
+
+// TODO(Sarah): Use constants to map ncr.status to bootstrap package status according to Apple
+// specs. https://developer.apple.com/documentation/devicemanagement/installenterpriseapplicationresponse
+func (ds *Datastore) GetHostMDMMacOSSetup(ctx context.Context, hostID uint) (*fleet.HostMDMMacOSSetup, error) {
+	// TODO(Sarah): Is ncr.result the correct column to use here? I don't see where error details are stored.
+	stmt := `
+SELECT
+    CASE 
+        WHEN ncr.status = 'Acknowledged' THEN 'installed'
+        WHEN ncr.status = 'Error' THEN 'failed'
+        ELSE 'pending' 
+    END AS bootstrap_package_status,
+    COALESCE(ncr.result, '') AS detail
+FROM
+    hosts h
+JOIN host_mdm_apple_bootstrap_packages hmabp ON
+    hmabp.host_uuid = h.uuid
```

Comment:
Same question as above regarding uniqueness of `hmabp` for a given host.

Reply:
technically no, but I think you're right and it's a good idea to be cautious here.

Reply:
@roperzh, do you think we should just make `host_uuid` alone the primary key for `hmabp` rather than `host_uuid + command_uuid`?